course/src/isometry.cc

```diff
@@ -0,0 +1,141 @@
+const Vector3 Vector3::kZero = {0., 0., 0.};
+
+Vector3 Vector3::operator+(const Vector3 &other) const {
+  return Vector3(x_ + other.x_, y_ + other.y_, z_ + other.z_);
+}
+
+Vector3 Vector3::cross(const Vector3 &other) const {
+  Vector3 output;
+  output.x_ = y() * other.z() - z() * other.y();
+  output.y_ = z() * other.x() - x() * other.z();
+  output.z_ = x() * other.y() - y() * other.x();
+  return output;
+}
+
+Vector3 Vector3::operator/(const Vector3 &other) const {
+  return Vector3(x_ / other.x_, y_ / other.y_, z_ / other.z_);
+}
+
+Vector3 Vector3::operator-(const Vector3 &other) const {
+  return Vector3(x_ - other.x_, y_ - other.y_, z_ - other.z_);
+}
+
+Vector3 Vector3::operator*(const Vector3 &other) const {
+  return Vector3(x_ * other.x_, y_ * other.y_, z_ * other.z_);
+}
+
+Vector3 Vector3::operator*(const double &other) const {
+  return Vector3(x_ * other, y_ * other, z_ * other);
+}
+
+bool Vector3::operator==(const Vector3 &other) const {
+  return (x_ == other.x_ && y_ == other.y_ && z_ == other.z_);
+}
+
+Vector3::Vector3(const std::initializer_list<double> &rhs) {
+  auto begin_ptr = rhs.begin();
+  x_ = *(begin_ptr++);
+  y_ = *(begin_ptr++);
+  z_ = *(begin_ptr);
+}
+double Vector3::norm() const { return (sqrt(x_ * x_ + y_ * y_ + z_ * z_)); }
+
+double Vector3::dot(const Vector3 &other) const {
+  return (x_ * other.x_ + y_ * other.y_ + z_ * other.z_);
+}
+
+bool Vector3::operator==(const std::initializer_list<double> &rhs) const {
+  Vector3 vec{rhs};
+  return *this == vec;
+}
+
+double &Vector3::operator[](const int index) {
+  switch (index) {
+  case 0:
+    return x_;
+
+  case 1:
+    return y_;
+
+  default:
+    return z_;
+  }
+}
+
+const double &Vector3::operator[](const int index) const {
+  switch (index) {
+  case 0:
+    return x_;
+
+  case 1:
+    return y_;
+
+  default:
+    return z_;
+  }
+}
+
+Matrix3::Matrix3(const std::initializer_list<double> &rhs) {
+  auto begin_vec = rhs.begin();
+  _matrix[0] = Vector3{begin_vec[0], begin_vec[1], begin_vec[2]};
+  _matrix[1] = Vector3{begin_vec[3], begin_vec[4], begin_vec[5]};
+  _matrix[2] = Vector3{begin_vec[6], begin_vec[7], begin_vec[8]};
+}
+
+bool Matrix3::operator==(const Matrix3 &other) const {
+  return (_matrix[0] == other._matrix[0] && _matrix[1] == _matrix[1] &&
+          _matrix[2] == other._matrix[2]);
+}
+
+Matrix3 Matrix3::operator-(const Matrix3 &other) const {
+  return (Matrix3(_matrix[0] - other._matrix[0], _matrix[1] - other._matrix[1],
+                  _matrix[2] - _matrix[2]));
+}
+
+Matrix3 Matrix3::operator+(const Matrix3 &other) const {
+  return (Matrix3(_matrix[0] + other._matrix[0], _matrix[1] + other._matrix[1],
+                  _matrix[2] + _matrix[2]));
+}
+
+Matrix3 Matrix3::operator*(const double &other) const {
+  return (Matrix3(_matrix[0] * other, _matrix[1] * other, _matrix[2] * other));
+}
+
+Matrix3 Matrix3::operator*(const Matrix3 &other) const {
+  Matrix3 output;
+  for (int i = 0; i < 3; i++)
```

Comment:
You're storing rows as vector, and you already have the operation vector*vector. Use it.

Reply:
Addressed